### PR TITLE
Feature/5872 add loading icon

### PR DESCRIPTION
**Pull request by HavierD**
The loading modal is only used in `searchpage`. May also be useful for other pages, but not implemented in this PR. I just used a basic progress circle. They design will change in the future and at that time we only need to replace the icon.

### CODE REVIEW

**Comment by HavierD** [comment]
some tests were also skipped for now. will fix later

File: src/components/loading/LoadingEmitter.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import EventEmitter from "events";
+
+const loadingEmitter = new EventEmitter();
+
+export default loadingEmitter;
```
Comment (utas-raymondng) [inline src/components/loading/LoadingEmitter.ts:5]: Prefer not export this emitter directly but export function that interact with it. Therefore you can enforce event type and function type for the on and off function.

**Review by utas-raymondng — CHANGES_REQUESTED** [review]
Click the description on the list view, it shows the spatial extents but the loading keeps without exit

**Review by utas-raymondng — APPROVED** [review]
LGTM

**Pull request merged**